gradient_descent.js

```diff
@@ -1,11 +1,36 @@
-class GradientDescent {
-  constructor(objective, h = 0.01, x_ini = [1,2], delta = 0.1) {
-    this.h = h;
+/** Mother class algorithm must do the following tasks
```

Comment:
```suggestion
/** Abstract class Algorithm is only used to define the following API for implemented algorithms:
```

Reply:
In the future we could use typescript to be able to define the interfaces more precisely but it's good for now :)